libraries/botbuilder-core/botbuilder/core/teams/teams_activity_handler.py

```diff
@@ -0,0 +1,404 @@
+            if member.additional_properties != {}:
+                team_members_added.append(TeamsChannelAccount(member))
+            else:
+                if team_members == {}:
+                    result = await TeamsInfo.get_members_async(turn_context)
+                    team_members = { i.id : i for i in result }
+
+                if member.id in team_members:
+                    team_members_added.append(member)
+                else:
+                    newTeamsChannelAccount = TeamsChannelAccount(
+                        id=member.id,
+                        name = member.name,
+                        aad_object_id = member.aad_object_id,
+                        role = member.role
+                        )
+                    team_members_added.append(newTeamsChannelAccount)
+
+        return await self.on_teams_members_added_activity(teams_members_added, team_info, turn_context)
+        """
+        for member in members_added:
+            new_account_json = member.seralize()
+            del new_account_json["additional_properties"]
+            member = TeamsChannelAccount(**new_account_json)
+        return await self.on_teams_members_added_activity(members_added, turn_context)
+
+    async def on_teams_members_added_activity(
+        self, teams_members_added: [TeamsChannelAccount], turn_context: TurnContext
+    ):
+        for member in teams_members_added:
+            member = ChannelAccount(member)
+        return super().on_members_added_activity(teams_members_added, turn_context)
+
+    async def on_teams_members_removed_dispatch_activity(  # pylint: disable=unused-argument
+        self,
+        members_removed: [ChannelAccount],
+        team_info: TeamInfo,
+        turn_context: TurnContext,
+    ):
+        teams_members_removed = []
+        for member in members_removed:
+            new_account_json = member.seralize()
+            del new_account_json["additional_properties"]
+            teams_members_removed.append(TeamsChannelAccount(**new_account_json))
+
+        return await self.on_teams_members_removed_activity(
+            teams_members_removed, turn_context
+        )
+
+    async def on_teams_members_removed_activity(
+        self, teams_members_removed: [TeamsChannelAccount], turn_context: TurnContext
+    ):
+        members_removed = [ChannelAccount(i) for i in teams_members_removed]
+        return super().on_members_removed_activity(members_removed, turn_context)
+
+    async def on_teams_channel_deleted_activity(  # pylint: disable=unused-argument
+        self, channel_info: ChannelInfo, team_info: TeamInfo, turn_context: TurnContext
+    ):
+        return  # Task.CompleteTask
+
+    async def on_teams_channel_renamed_activity(  # pylint: disable=unused-argument
+        self, channel_info: ChannelInfo, team_info: TeamInfo, turn_context: TurnContext
+    ):
+        return  # Task.CompleteTask
+
+    async def on_teams_team_reanamed_async(  # pylint: disable=unused-argument
```

Comment:
Probably can push this to the issue where you rename the methods.  But "async" wouldn't be consistent?  Also, spelling error.